src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Plugin.Program.Programs;
+using Wox.Infrastructure.Logger;
+using Wox.Infrastructure.Storage;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Plugin.Program.Storage
+{
+    using Win32 = Programs.Win32;
+    internal class Win32ProgramRepository : ListRepository<Programs.Win32>, IProgramRepository
+    {
+        private IStorage<IList<Programs.Win32>> _storage;
+        private Settings _settings;
+        private List<IFileSystemWatcherWrapper> _fileSystemWatcherHelpers;
+        private string[] _pathsToWatch;
+        private int _numberOfPathsToWatch;
+        private Collection<string> extensionsToWatch = new Collection<string>{ "*.exe", "*.lnk", "*.appref-ms", "*.url" };
+        private readonly string lnkExtension = ".lnk";
+        private readonly string urlExtension = ".url";
+
+        public Win32ProgramRepository(List<IFileSystemWatcherWrapper> fileSystemWatcherHelpers, IStorage<IList<Win32>> storage, Settings settings, string[] pathsToWatch)
```

Comment:
nit: this could be slightly more flexible if it takes `IList<IFileSystemWatcherWrapper>`